File: app/src/main/java/com/fitaleks/heroesthesaurus/data/source/CharactersDataSource.java
package com.fitaleks.heroesthesaurus.data.source;

import android.support.annotation.NonNull;

import com.fitaleks.heroesthesaurus.data.Character;

import java.util.List;

import rx.Observable;

/**
 * Created by Alexander on 01.12.16.
 */

public interface CharactersDataSource {
    Observable<List<Character>> getCharacters();
    Observable<Character> getCharacter(@NonNull String characterId);
    Observable<List<Character>> searchForCharacters(@NonNull final String query);

    void deleteAllCharacters();
    void saveCharacter(Character character);
    void refreshCharacters();

}

File: app/src/main/java/com/fitaleks/heroesthesaurus/data/source/CharactersDataSource.kt
package com.fitaleks.heroesthesaurus.data.source

import com.fitaleks.heroesthesaurus.data.MarvelCharacter

/**
 * Created by Alexander on 01.12.16.
 */
interface CharactersDataSource {
//    fun getCharacters(): Observable<List<MarvelCharacter>>
//    fun getCharacter(characterId: String): Observable<MarvelCharacter>?
//    fun searchForCharacters(query: String): Observable<List<MarvelCharacter>>

    fun deleteAllCharacters()
    fun saveCharacter(character: MarvelCharacter)
    fun refreshCharacters()
}
